pippy/PipelineStage.py

```diff
@@ -485,6 +486,36 @@ def _send_grads(
         return grad_send_reqs
 
     def forward(self, *args, **kwargs):
+        # If submod is wrapped with DDP, we use the `no_sync` context manager to
+        # avoid gradient all-reduce per microbatch
+        def forward_maybe_with_ddp(*args, **kwargs):
```

Comment:
Probably change the name with `forward_maybe_with_nosync`. We can use the same logic for FSDP.